bindings/python/CMakeLists.txt

```diff
@@ -95,7 +95,7 @@ else()
     CODE
     "EXECUTE_PROCESS(
       COMMAND /usr/bin/env ${XROOTD_PYBUILD_ENV} ${PYTHON_EXECUTABLE} -m pip install \
-                ${PIP_OPTIONS} \
+                --ignore-installed ${PIP_OPTIONS} \
```

Comment:
`--ignore-installed` is almost NEVER what you want to do and certainly NOT what you want to do by default. This should not go in as is.

Reply:
Sorry the all caps are a bit much &mdash; I was in a rush while typing this and didn't think to leave a more thoughtful response. The reason why I would suggest trying to always avoid `--ignore-installed` is laid out in the docs which I quote there. As it name suggests, you're literally telling `pip` to act like you're installing into a blank directory space that has just been created. This can cause obvious pathologies as you're now subject to having potentially anything in the the package space that can lead to confusing and undesirable behavior at runtime.

If you intentionally are using `--ignore-installed` for a situation where [you know what you're doing and there really isn't any better alternative](https://github.com/matthewfeickert/cvmfs-venv/blob/afd316674c954b5f709bf7a0181899b8045f3463/README.md#disadvantages) then that's obviously fine, but this should definitely not be the default behavior for anything.

This is why I suggested in https://github.com/xrootd/xrootd/issues/1768#issuecomment-1224217117 to just use the `PIP_OPTIONS` CMake option which was added for situations like this.

Reply:
No worries. I actually sort of agree with the above, and that's [what I did](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=769b1f2e6faabebc6d0900142cf6d671db1e9ed3) in Gentoo to get the package fixed. However, I think that maybe having an alternative to using the pip install method altogether would be better. If someone just wants to make a binary tarball of XRootD to install elsewhere and tries the usual `make DESTDIR=xrootd-5.x/ install`, that may, as a side effect, just remove the currently installed version of XRootD from their system. This is probably not a good idea as default behavior and why I thought that having that `--ignore-installed` as default would be better (also since `make install` usually has as default behavior installation without asking/uninstalling anything). That being said, I didn't see an option that can negate it by adding to `PIP_OPTIONS` so maybe a different solution is preferable. One idea would be to have `PIP_OPTIONS="--user"` by default, for example, to avoid the situation I describe above.

Reply:
> If someone just wants to make a binary tarball of XRootD to install elsewhere and tries the usual make `DESTDIR=xrootd-5.x/` install, that may, as a side effect, just remove the currently installed version of XRootD from their system.

If that's the situation then [`pip`'s `--target` option](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-t) should be used

> Install packages into `<dir>`. By default this will not replace existing files/folders in `<dir>`. Use `--upgrade` to replace existing packages in `<dir>` with new versions.

Example:

```console
$ docker run --rm -ti python:3.10 /bin/bash
root@0abcc4f646be:/# python -m venv venv && . venv/bin/activate
(venv) root@0abcc4f646be:/# python -m pip --quiet install --upgrade pip
(venv) root@0abcc4f646be:/# python -m pip install 'numpy<1.23.0'
Collecting numpy<1.23.0
  Downloading numpy-1.22.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.8/16.8 MB 23.1 MB/s eta 0:00:00
Installing collected packages: numpy
Successfully installed numpy-1.22.4
(venv) root@0abcc4f646be:/# mkdir /example
(venv) root@0abcc4f646be:/# python -m pip --quiet install --target /example --upgrade numpy
(venv) root@0abcc4f646be:/# python -m pip show numpy
Name: numpy
Version: 1.22.4
Summary: NumPy is the fundamental package for array computing with Python.
Home-page: https://www.numpy.org
Author: Travis E. Oliphant et al.
Author-email: 
License: BSD
Location: /venv/lib/python3.10/site-packages
Requires: 
Required-by: 
(venv) root@0abcc4f646be:/# ls /example/
bin  numpy  numpy-1.23.2.dist-info  numpy.libs
(venv) root@0abcc4f646be:/#
```

Trying that instead of the use of `--prefix` might be a reasonable approach.

> That being said, I didn't see an option that can negate it by adding to `PIP_OPTIONS` so maybe a different solution is preferable.

Just use `DPIP_OPTIONS="--ignore-installed"` for your particular use case.

> One idea would be to have `PIP_OPTIONS="--user"` by default, for example, to avoid the situation I describe above.

That is also something that should not be done. `--user` should not be used unless there are extreme situations as you should not be installing packages with `pip` globally.

Reply:
I had looked into the `--root` option only, I think indeed the `--target` option is what we need here. I will update my pull request using that instead of `--ignore-installed`.

Reply:
> I had looked into the `--root` option only, I think indeed the `--target` option is what we need here. I will update my pull request using that instead of `--ignore-installed`.

No. You will need to add it as an option in [`bindings/python/CMakeLists.txt`](https://github.com/xrootd/xrootd/blob/ef4f041db001d791adb14d42cbc51d84bcd82117/bindings/python/CMakeLists.txt) and not as a global default.